app/src/end2endTest/java/gov/va/vro/end2end/VroV2Tests.java

```diff
@@ -32,6 +32,23 @@ public class VroV2Tests {
 
   private final RestTemplate restTemplate = new RestTemplate();
 
+  /*
+   * This test checks the RequestBodyAdvice sanitizing logic for disallowed characters.
+   * Eventually we should refactor this out into its own test suite with other endpoints
+   * for any security-related HTTP tests.
+   */
+  @Test
+  void testExamOrderingStatus_disallowedCharacters() {
+    var request = getOrderingStatusDisallowedCharacters();
+    var requestEntity = getEntity(request);
+    try {
+      restTemplate.postForEntity(EXAM_ORDERING_STATUS_URL, requestEntity, String.class);
+      fail("Should have thrown exception");
+    } catch (Exception e) {
+      assertTrue("400 : \"{\"message\":\"Bad Request\"}\"".equals(e.getMessage()));
+    }
```

Comment:
I am confused by this test. Why would a post raise an exception? Isn't this testing the API? We should get a response object and check for 400?

Reply:
It looks like it is because our end2endTest is embedded within the app itself (`/app/src/end2endTest/...`). So when an exception is thrown, within the app and not explicitly caught, the HTTP request fails and the test bails.

This is a pattern I repeated from a previous example `testExamOrderingStatus_invalidRequest()`

I think in order to fix this properly, the end2endTest would need to be logically separated from the rest of the app into its own (`/end2endTest/...`) -- which would allow us to run the test suite independent of the app and simply does HTTP requests and checks the responses. Right now this appears to be more of a coupled and integrated end2end testing suite.